diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -301,6 +303,16 @@ fun ASTNode?.isAccessibleOutside(): Boolean =
             true
         }
 
+fun ASTNode.hasSuppress(warningName: String): Boolean {
+    if (elementType == FILE) {
+        return findAllNodesWithSpecificType(ANNOTATION_ENTRY).any {
+            it.text.contains("Suppress") && it.text.contains(warningName)
+        }
+    }
+    return parent({ it.findAllNodesWithSpecificType(ANNOTATION_ENTRY).any {
+        it.text.contains("Suppress") && it.text.contains(warningName)
```

Comment:
It is incorrect logic, there can be `@Supresss` annotation and it will be ignore.